lighthouse-core/audits/accessibility/nested-interactive.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * @license Copyright 2021 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/**
+ * @fileoverview Ensure interactive controls are not nested.
+ * See base class in axe-audit.js for audit() implementation.
+ */
+
+const AxeAudit = require('./axe-audit.js');
+const i18n = require('../../lib/i18n/i18n.js');
+
+const UIStrings = {
+  /** Title of an accesibility audit that checks if there are any nested interactive elements on the page. This title is descriptive of the successful state and is shown to users when no user action is required. */
+  title: 'Interactive controls are not nested',
+  /** Title of an accesibility audit that checks if there are any nested interactive elements on the page. This title is descriptive of the failing state and is shown to users when there is a failure that needs to be addressed. */
+  failureTitle: 'Interactive controls are nested',
+  /** Description of a Lighthouse audit that tells the user *why* they should try to pass. This is displayed after a user expands the section to see more. No character length limits. 'Learn More' becomes link text to additional documentation. */
+  description: 'Nested interactive controls will not be announced by assistive technologies such as screen readers. [Learn more.](https://dequeuniversity.com/rules/axe/4.2/nested-interactive)',
+};
+
+const str_ = i18n.createMessageInstanceIdFn(__filename, UIStrings);
+
+class NestedInteractive extends AxeAudit {
```

Comment:
if `nested-interactive` is problematic, I say we just put `'nested-interactive': {enabled: false},` in `accessibility.js` and put a TODO linked to your https://github.com/dequelabs/axe-core/issues/2958. Looking at the history of `nested-interactive`, the discussion seems to be all around  `<button>` and e.g. links inside elements that are `role="checkbox"`, but then it became _all_ presentational elements. We thought we were doing well in the report with our `<details>`, so I think in this case we at least need to figure out how we fix ourselves before we demand other people fix their own sites :)